pkg/target/target_test.go

```diff
@@ -430,3 +438,144 @@ func TestProcessData(t *testing.T) {
 		})
 	}
 }
+
+var fooMatch = match.Match{
+	Kinds: []match.Kinds{
+		{
+			Kinds:     []string{"Thing"},
+			APIGroups: []string{"some"},
+		},
+	},
+	Scope:              "Namespaced",
+	Namespaces:         []util.Wildcard{"my-ns"},
+	ExcludedNamespaces: nil,
+	LabelSelector: &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"obj": "label",
+		},
+	},
+	NamespaceSelector: &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"ns": "label",
+		},
+	},
+	Name: "",
+}
+
+var fooConstraint = makeConstraint(
+	setKinds([]string{"some"}, []string{"Thing"}),
+	setScope("Namespaced"),
+	setNamespaceName("my-ns"),
+	setLabelSelector("obj", "label"),
+	setNamespaceSelector("ns", "label"),
+)
+
+func TestToMatcher(t *testing.T) {
+	unstructAssign, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(&unversioned.Assign{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec: unversioned.AssignSpec{
+			Match: fooMatch,
+		},
+	})
+	tests := []struct {
+		name       string
+		constraint *unstructured.Unstructured
+		want       constraints.Matcher
+		wantErr    bool
+	}{
+		{
+			name:       "constraint with no match fields",
+			constraint: makeConstraint(),
+			wantErr:    true,
+		},
+		{
+			name:       "constraint with match fields",
+			constraint: fooConstraint,
+			want: &Matcher{
+				&fooMatch,
+			},
+		},
+		{
+			name: "mutator with match fields",
+			constraint: &unstructured.Unstructured{
+				Object: unstructAssign,
+			},
+			want: &Matcher{
+				&fooMatch,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := &K8sValidationTarget{}
+			got, err := h.ToMatcher(tt.constraint)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ToMatcher() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
Use cmp.Diff and print the resulting diff. If you have problems using cmp.Diff for this case - ask me (I'm super familiar with it)

Reply:
Done.